app/src/shared/components/ActivityList/index.jsx

```diff
@@ -0,0 +1,118 @@
+// @flow
+
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import { useSelector, useDispatch } from 'react-redux'
+
+import Activity, { actionTypes } from '$shared/utils/Activity'
+import { getFilters } from '$userpages/utils/constants'
+import { selectStreams } from '$userpages/modules/userPageStreams/selectors'
+import { getStreams } from '$userpages/modules/userPageStreams/actions'
+import { selectMyProductList } from '$mp/modules/myProductList/selectors'
+import { getMyProducts } from '$mp/modules/myProductList/actions'
+import { selectCanvases } from '$userpages/modules/canvas/selectors'
+import { getCanvases } from '$userpages/modules/canvas/actions'
+
+import ActivityListItem from './ActivityListItem'
+
+const Container = styled.div`
+    width: 400px;
+    height: 560px;
+    display: flex;
+    flex-direction: column;
+`
+
+const Tabs = styled.div`
+    display: flex;
+    width: 100%;
+    padding: 16px 0 0 16px;
+`
+
+const Tab = styled.div`
+    display: flex;
+    font-size: 14px;
+    font-weight: 500;
+    line-height: 16px;
+    color: #323232;
+    opacity: ${(props) => (props.active ? '1' : '0.5')};
+    border-bottom: ${(props) => (props.active ? '1px solid #323232' : 'none')};
+    margin-right: 40px;
+    cursor: pointer;
+    padding-bottom: 16px;
+`
+
+const Items = styled.div`
+    width: 100%;
+    height: 100%;
+    overflow: auto;
+`
+
+type Props = {
+    activities: Array<Activity>,
+}
+
+const notificationFilter = (item: Activity) => (
+    (item.action === actionTypes.PAYMENT || item.action === actionTypes.SHARE)
+)
+
+const activityFilter = (item: Activity) => (
+    !notificationFilter(item)
+)
+
+const ActivityList = ({ activities }: Props) => {
+    const dispatch = useDispatch()
+    const [activeTab, setActiveTab] = useState('activity')
+    const streams = useSelector(selectStreams)
+    const products = useSelector(selectMyProductList)
+    const canvases = useSelector(selectCanvases)
+
+    if (activities.some((n) => n.streamId != null) && (streams == null || streams.length === 0)) {
```

Comment:
What if it's a really long list? Do we limit what we display in any way?

Reply:
Yes, we load activity from only last 30 days. It can still be long list though :/